src/PointwiseFunctions/AnalyticSolutions/Tags.hpp

```diff
@@ -80,18 +73,231 @@ struct Error : db::PrefixTag, db::SimpleTag {
 /// \see ::Tags::AnalyticSolutions
 struct AnalyticSolutionsBase : db::BaseTag {};
 
-/// The analytic solutions for all `FieldTags`
+namespace detail {
+template <typename Tag>
+struct AnalyticImpl : db::PrefixTag, db::SimpleTag {
+  using type = typename Tag::type;
+  using tag = Tag;
+};
+}  // namespace detail
+
 template <typename FieldTags>
-struct AnalyticSolutions : AnalyticSolutionsBase, db::SimpleTag {
-  using type = ::Variables<db::wrap_tags_in<Analytic, FieldTags>>;
+struct AnalyticSolutions : ::Tags::AnalyticSolutionsBase, db::SimpleTag {
+  using field_tags = FieldTags;
+  using type = std::optional<
+      ::Variables<db::wrap_tags_in<detail::AnalyticImpl, FieldTags>>>;
+};
+
+template <typename Tag>
+struct is_analytic_solutions : std::false_type {};
```

Comment:
Please add a comment what you need this for. I suggest to name this `is_analytic_solutions_tag`.

Reply:
it's already in the `Tags` namespace, so I don't see a reason to add more qualifiers to the name.